ansible-ipi-install/roles/installer/tasks/25c_disconnected_registry_create.yml

```diff
@@ -0,0 +1,186 @@
+---
+
+- name: Create a disconnected registry
+  block:
+    - name: Make sure needed packages are installed
+      package:
+        state: present
+        name: "{{ item }}"
+      loop:
+        - "{{ packages_registry | default([]) }}"
+      become: true
+    
+    - name: open firewall port {{ registry_port }}
+      firewalld:
+        port: "{{ registry_port }}/tcp"
+        permanent: yes
+        immediate: yes
+        state: enabled
+      become: true
+    
+    - name: query getent for homedir
+      getent:
+        database: passwd
+        key: "{{ ansible_ssh_user }}"
+    
+    - name: set homedir fact
+      set_fact:
+        homedir: "{{ getent_passwd | json_query('*[4]') | first }}"
+    
+    - name: Create directory to hold the registry files
+      file:
+        path: "{{ item }}"
+        state: directory
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+        mode: u=rwX,g=rX,o=rX
+        recurse: yes
+      loop:
+        - "{{ registry_dir_auth }}"
+        - "{{ registry_dir_cert }}"
+        - "{{ registry_dir_data }}"
+      become: true
+    
+    - name: Generate htpasswd entry
+      command: htpasswd -bBn dummy dummy
```

Comment:
would be great if user/password are set with variables.

Reply:
i have included this and it will be pushed with my next commit

Reply:
pushed